util/topgen.py

```diff
@@ -793,6 +793,8 @@ def generate_top_ral(top: Dict[str, object],
         inst_name = module['name']
         block_name = module['type']
         block = name_to_block[block_name]
+        if "attr" in module:
+            block.attr = module["attr"]
```

Comment:
This isn't right: the `IpBlock` object (`block`) describes the type of block that gets instantiated. Here, you're updating it with a value that depends on a particular instantiation.

Reply:
I don't know what the best way to pass this `attr` to reggen. any suggestion?
https://github.com/lowRISC/opentitan/blob/8e56cfc0688e07bac071b46126b2317e202e6bd2/hw/top_earlgrey/data/top_earlgrey.hjson#L460

Reply:
Oh, wait! I thought you were adding it to the file format that reggen parses. Now I understand what you're doing. The best thing to do is probably to add this to the `Top` class, which gets constructed by topgen and represents the instances in the top-level. There's probably scope for some refactoring here, but maybe that can wait for a later PR.

I'd also suggest rebasing the change on top of PR #5564 (which changes this code quite a bit, so there's going to be some painful rebasing otherwise).

Reply:
How about create this Dict to store the attr based on block name instance in `Top` class and use in top_uvm_reg.sv.tpl. WDYT?
`block_attr = {} # block instance name to attr, type: Dict[str, str]`